lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -46,6 +46,18 @@ bool ExportVerilog::isSimpleReadOrPort(Value v) {
   return isa<WireOp, RegOp>(readSrc);
 }
 
+// Check if the value is deemed worth spilling into a wire.
+static bool shouldSpillWire(Operation &op, const LoweringOptions &options) {
+  auto isAssign = [](Operation *op) { return isa<AssignOp>(op); };
+
+  // If there are more than the maximum number of terms in this single result
+  // expression, and it hasn't already been spilled, this should spill.
+  return isVerilogExpression(&op) &&
+         op.getNumOperands() > options.maximumNumberOfTermsPerExpression &&
+         op.getNumResults() == 1 &&
+         llvm::none_of(op.getResult(0).getUsers(), isAssign);
+}
```

Comment:
I like how this creates a nice point to implement additional heuristics later to decide when to spill things into more wires. :+1:

We could later do things like estimating the character length of the expression recursively and then deciding whether we want to split/spill.